### PR TITLE
Switch PyProxy.toString from repr to str

### LINKED ISSUE #4040: Question: Should `toString` use `__repr__` or `__str__`?
In https://github.com/pyodide/pyodide/issues/4037#issuecomment-1665679197 @Duncanr-glitch said:

> `the toString` method using the ```__repr__``` method just seems incorrect since there's a distinction between ```__repr__``` and ```__str__``` that python devs expect and sometimes rely on. Using ```__repr__``` when doing a bare console.log makes since, but not when a dev is trying to string the object.

Looking at the history, `toString` was added here: https://github.com/pyodide/pyodide/commit/9579aa99bcfdecf65eb99311e6bb25fe8af831a9
I am not aware that we've ever discussed what this should do. I personally don't have a strong opinion. Interestingly, `toString` is missing from our proxy table, we should probably update the table in any case.
https://pyodide.org/en/stable/usage/type-conversions.html#proxying-from-python-into-javascript

@rth @ryanking13 @antocuni what do you all think?

### PR DESCRIPTION
**Pull request by hoodmane**
Resolves #4040.

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by hoodmane** [comment]
Astonishingly, it looks like we didn't have a single unit test that cares about this??

**Comment by hoodmane** [comment]
I added more string related entries to the `type-conversions.md`. There are a lot of things missing from `type-conversions.md`, I will hopefully go through and update it when I have a moment.

File: src/js/pyodide.ts [hunk 1]
```diff
@@ -395,6 +401,9 @@ export async function loadPyodide(
   if (Module.exited) {
     throw Module.exited.toThrow;
   }
+  if (options.pyproxyToStringRepr) {
+    API.setPyProxyToStringMethod("repr");
```
Comment (ryanking13) [inline src/js/pyodide.ts:405]: Why are we entering a string value rather than a boolean value here, and converting it inside the function?

**Review by ryanking13 — APPROVED** [review]
Thanks!

File: src/js/pyodide.ts (hunk 1 above)
Comment (hoodmane) [reply src/js/pyodide.ts:405]: Meant to say `true` here.

Comment (hoodmane) [reply src/js/pyodide.ts:405]: I'm surprised typescript is okay with this...

**Pull request merged**